mne_bids/tests/test_mne_bids.py

```diff
@@ -151,11 +164,31 @@ def test_vhdr():
 
     raw_to_bids(subject_id=subject_id, session_id=session_id, run=run,
                 task=task, raw_file=raw_fname, output_path=output_path,
-                overwrite=True, kind='eeg')
+                overwrite='overwrite', kind='eeg')
 
     cmd = ['bids-validator', '--bep006', output_path]
     run_subprocess(cmd, shell=shell)
 
+    # check the 'clear' overwrite command works correctly
+    raw_to_bids(subject_id=subject_id, session_id=session_id, run=run2,
+                task=task, raw_file=raw_fname, output_path=output_path,
+                overwrite='clear', kind='eeg')
+    # we will need to go over the tree to ensure there are no files with
+    # `run-01` in their file names:
+    for _, _, files in os.walk(output_path):
+        for f in files:
+            if 'run-%s' % run in f:
+                raise OSError(EEXIST)
```

Comment:
I'm a little confused by this test. First of all, you should not raise errors in tests. Rather you should assert stuff. Secondly, I think from what I understood, "clear" should remove everything -- not just files with `run-01` ...

Reply:
I deliberated this one a little bit.
The logic here is that I am testing to see if there are any files from the previous run of the script in the session level folder (as this is what the clear applies to).
I tried a few different ways of making this clean, however we want the test to pass if no error is raised here. Because of this you can't use `pytest.raises` because if no error is raised it will cause an error.
From what i could tell from googling the issue a bit the agreed way is for an issue to be raised, because then your test is failing. Lack of error raised is the test passing so all is good.
It doesn't feel as nice but there didn't seem to be a better way without writing some function which essentially is a `not_raised` function.

Reply:
> however we want the test to pass if no error is raised here

I don't get this part. Can't you just assert that the number of files in the folder is equal to 0?

Reply:
But there will be files in the folder. The only time there won't be any is right after the `make_bids_folders` function runs, and we can't get a test to run at this time.
That's why I check to make sure that the files from the previous run do not exist

Reply:
Oh I see. Then shouldn't you be testing `make_bids_folders` directly? Call `raw_to_bids` and then `make_bids_folders` with `'clear'`. Then you can assert that the number of files in the folder is equal to 0

Reply:
@monkeyman192 what was the conclusion here?